MdePkg/Include/Protocol/MediaSanitize.h

```diff
@@ -0,0 +1,147 @@
+                                  protection.
+  @retval EFI_DEVICE_ERROR        The device reported an error while attempting
+                                  to perform the sanitize operation.
+  @retval EFI_INVALID_PARAMETER   The clear request contains parameters that
+                                  are not valid.
+  @retval EFI_NO_MEDIA            There is no media in the device.
+  @retval EFI_MEDIA_CHANGED       The MediaId is not for the current media.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EFI_BLOCK_MEDIA_CLEAR)(
+  IN     EFI_MEDIA_SANITIZE_PROTOCOL   *This,
+  IN     UINT32                        MediaId,
+  IN     UINT32                        PassCount,
+  IN     VOID                          *SectorOwBuffer
+  );
+
+/**
+  Purge Media utilizes native Sanitize operations. Transport specific
+  overwrite, block erase, or crypto erase functions shall be invoked based
+  on transport.
+
+  @param[in] This             Indicates a pointer to the calling context.
+  @param[in] MediaId          The media ID that the clear request is for.
+  @param[in] PurgeAction      Sanitize action: overwrite, crypto or block erase.
+  @param[in] OverwritePattern 32-bit pattern to overwrite on media.
+
+  @retval EFI_SUCCESS             The sanitize request was queued if Event is
+                                  not NULL. The data was erased correctly to the
+                                  device if the Event is NULL.to the device.
+  @retval EFI_WRITE_PROTECTED     The device can't be sanitized due to write
+                                  protection.
+  @retval EFI_DEVICE_ERROR        The device reported an error while attempting
+                                  to perform the sanitize operation.
+  @retval EFI_INVALID_PARAMETER   The clear request contains parameters that
+                                  are not valid.
+  @retval EFI_NO_MEDIA            There is no media in the device.
+  @retval EFI_MEDIA_CHANGED       The MediaId is not for the current media.
+
+**/
+typedef
+EFI_STATUS
+(EFIAPI *EFI_BLOCK_MEDIA_PURGE)(
+  IN     EFI_MEDIA_SANITIZE_PROTOCOL   *This,
+  IN     UINT32                        MediaId,
+  IN     UINT32                        PurgeAction,
+  IN     UINT32                        OverwritePattern
+  );
+
+///
+/// The EFI Media Sanitize Protocol provides the ability for a device to expose
+/// sanitize functionality. This optional protocol is installed on the same handle
+/// as the EFI_BLOCK_IO_PROTOCOL or EFI_BLOCK_IO2_PROTOCOL.
+///
+struct _EFI_MEDIA_SANITIZE_PROTOCOL {
+  ///
+  /// The revision to which the EFI_MEDIA_SANITIZE_PROTOCOL adheres. All future
+  /// revisions must be backwards compatible. If a future version is not
+  /// backwards compatible, it is not the same GUID.
+  ///
+  UINT64                       Revision;
+
+  ///
+  /// A pointer to the EFI_BLOCK_IO_MEDIA data for this device.
+  /// Type EFI_BLOCK_IO_MEDIA is defined in BlockIo.h.
+  ///
+  EFI_BLOCK_IO_MEDIA           *Media;

```

Comment:
Ultimately, I see the caller passing in a LUN Id or Namespace Id as the Media Id. The caller does not required to have the block io protocol, but may likely have it. 

I understand your point/concern however. If the caller has access to the actual Block IO or Media Sanitize protocol, then there is no need to pass in the Media Id. 

If the assumption can be made that Block IO is installed on every LUN, namespace, etc., then I'm fine removing Media Id as a parameter. The caller can match their tracked LUN, namespace, etc. on their end first.

Reply:
i wasn't trying to say the mediaId isn't a valid parameter or a good sanity check.  Seems like there is/might be a design where the block io changes content but protocol instance (memory location) stays the same.  In that case it seems reasonable for the caller to get the media id at the beginning when locating this and then use that for future calls to make sure they are still talking to the expected device?  I was just trying to understand the process here.

Reply:
That was actually the precise original intent :-). Matching of the expected media vs. any changes that may have happened in a different context or boot (i.e. like an LBA format change or hot swap).

Reply:
I am going to resolve this comment since it appears you understood my original intent. Please let me know if you'd like to go in a different direction. Thanks.